exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -1303,6 +1305,8 @@ private void checkGroupAndAggrValues(int incomingRowIdx) {
       long memDiff = allocator.getAllocatedMemory() - allocatedBeforeHTput;
       if ( memDiff > 0 ) { logger.warn("Leak: HashTable put() OOM left behind {} bytes allocated",memDiff); }
 
+      checkForSpillPossibility(currentPartition);
```

Comment:
Not sure this check 'chooseAPartitionToFlush'  is needed. If an exception is desired, I would think modifying doSpill() is better way e.g. modifying this line: "  if ( victimPartition < 0 ) { return; } " Otherwise in this process chooseAPartitionToFlush will be called twice.

      int victimPartition = chooseAPartitionToFlush(currentPartition, forceSpill);

      // In case no partition has more than one batch -- try and "push the limits"; maybe next
      // time the spill could work.
      if ( victimPartition < 0 ) { return; }

Reply:
These checks were needed to avoid infinite loop when there is not enough memory for the spill. 
I moved these checks into `spillIfNeeded()` method, so when called `doSpill()`, `forceSpill` in `spillIfNeeded()` is true and check should be done.